src/python/dxpy/utils/executable_unbuilder.py

```diff
@@ -115,14 +116,45 @@ def make_cluster_bootstrap_script_file(region, entry_point, code, suffix):
         Writes the string `code` into a file at the relative path
         "src/<region>_<entry_point>_clusterBootstrap.<suffix>"
         """
-        script_name = "src/%s_%s_clusterBootstrap.%s" % (region, entry_point, suffix)
+        script_name = "src/{}_{}_clusterBootstrap.{}".format(region, entry_point, suffix)
         with open(script_name, "w") as f:
             f.write(code)
         return script_name
-
-    # Get all the asset bundles
-    asset_depends = []
-    deps_to_remove = []
+    
+    # Get regions where the user's billTo are permitted
+    try:
+        bill_to = dxpy.api.user_describe(dxpy.whoami())['billTo']
+        permitted_regions = set(dxpy.DXHTTPRequest('/' + bill_to + '/describe', {}).get("permittedRegions"))
+    except DXError:
+        print("Failed to get permitted regions where {} can perform billable activities.\n".format(bill_to), file=sys.stderr)
+        sys.exit(1)    
+    
+    enabled_regions = set(info["runSpec"]["bundledDependsByRegion"].keys())
+    if not enabled_regions.issubset(permitted_regions):
+        print("Region(s) {} are not among the permitted regions of {}. Resources from these regions will not be available.".format(
+            ", ".join(enabled_regions.difference(permitted_regions)), bill_to))
+    # Update enabled regions
+    enabled_regions.intersection_update(permitted_regions)
+    # Check if at least one region is enabled when not omitting resources
+    if not omit_resources:
+        if not enabled_regions:
+            raise DXError(
+                "Cannot download resources of the requested executable {} since it is not available in any of the billable regions. "
+                "You can use the --omit-resources flag to skip downloading the resources. ".format(info["name"]))
+        
+        print("Dependencies could be retrieved from region: {}. ".format(", ".join(enabled_regions)))
+        # Pick a source region. The current selected region is preferred
+        try:
+            current_region = dxpy.api.project_describe(dxpy.WORKSPACE_ID, input_params={"fields": {"region": True}})["region"]
+        except:
+            current_region = None
+
+        if current_region in enabled_regions:
+            source_region  = current_region
+            print("Trying to download resources from the current region {}...".format(source_region))
+        else:
+            source_region = enabled_regions.pop()
+            print("Trying to download resources from one of the enabled region {}...".format(source_region))
 
     # When an applet is built bundledDepends are added in the following order:
```

Comment:
Please align it with the code (and place it above the code that that the comment is referring to)